app/models/application_choice.rb

```diff
@@ -12,6 +12,7 @@ class ApplicationChoice < ApplicationRecord
   has_one :accredited_provider, through: :course, class_name: 'Provider'
 
   has_many :notes, dependent: :destroy
+  validates :course_option, uniqueness: { scope: :application_form_id }
```

Comment:
Can we just add this as a test please 👍 

    it { is_expected.to validate_uniqueness_of(:course_option).scoped_to(:application_form_id) }

Reply:
Sorted